Enable the freedesktop.org backend on Android ("android" is replaced by a bogus
value in every cfg), same approach as for yazi.
The same replacement for Cargo.toml is done with sed in build.sh, since the
manifest is rewritten by `cargo vendor` and its exact layout can differ.

This dirty implementation of get_mount_points() for trash-rs for Android
has been tested and confirmed to work good enough to successfully move files to
trash in reverse dependency Yazi on Samsung Galaxy A70 SM-A705FN with LineageOS 20
Android 13 SELinux Enabled.

--- a/vendor/trash/src/freedesktop.rs
+++ b/vendor/trash/src/freedesktop.rs
@@ -815,6 +815,28 @@
     Ok(result)
 }
 
+#[cfg(target_os = "android")]
+fn get_mount_points() -> Result<Vec<MountPoint>, Error> {
+    // assume '/data' and '/strorage/emulated/0' are mounted
+    // for some reason, 'getmntent()' is not available in Rust on Android,
+    // despite being available in C on Android.
+    // https://github.com/rust-lang/libc/blob/0403150379d66216e90a0f8a804ae1ca33691067/src/unix/linux_like/android/mod.rs
+    let mut result = Vec::new();
+    let mount_point = MountPoint {
+        mnt_dir: "/data".into(),
+        _mnt_fsname: "/this/should/not/matter".to_string(),
+        _mnt_type: "ext4".to_string()
+    };
+    result.push(mount_point);
+    let mount_point = MountPoint {
+        mnt_dir: "/storage/emulated/0".into(),
+        _mnt_fsname: "/this/should/not/matter".to_string(),
+        _mnt_type: "fuse".to_string()
+    };
+    result.push(mount_point);
+    Ok(result)
+}
+
 #[cfg(any(target_os = "dragonfly", target_os = "freebsd", target_os = "openbsd"))]
 fn get_mount_points() -> Result<Vec<MountPoint>, Error> {
     use once_cell::sync::Lazy;
@@ -929,7 +951,8 @@
     target_os = "dragonfly",
     target_os = "freebsd",
     target_os = "openbsd",
-    target_os = "netbsd"
+    target_os = "netbsd",
+    target_os = "android"
 )))]
 fn get_mount_points() -> Result<Vec<MountPoint>, Error> {
     // On platforms that don't have support yet, return an error
--- a/vendor/trash/src/lib.rs
+++ b/vendor/trash/src/lib.rs
@@ -45,7 +45,7 @@
 #[path = "windows.rs"]
 mod platform;
 
-#[cfg(all(unix, not(target_os = "macos"), not(target_os = "ios"), not(target_os = "android")))]
+#[cfg(all(unix, not(target_os = "macos"), not(target_os = "ios"), not(target_os = "disabling_this_because_it_is_for_building_an_apk")))]
 #[path = "freedesktop.rs"]
 mod platform;
 
@@ -187,7 +187,7 @@
     },
 
     /// Error coming from file system
-    #[cfg(all(unix, not(target_os = "ios"), not(target_os = "android")))]
+    #[cfg(all(unix, not(target_os = "ios"), not(target_os = "disabling_this_because_it_is_for_building_an_apk")))]
     FileSystem {
         path: PathBuf,
         source: std::io::Error,
@@ -260,7 +260,7 @@
 impl error::Error for Error {
     fn source(&self) -> Option<&(dyn error::Error + 'static)> {
         match self {
-            #[cfg(all(unix, not(target_os = "macos"), not(target_os = "ios"), not(target_os = "android")))]
+            #[cfg(all(unix, not(target_os = "macos"), not(target_os = "ios"), not(target_os = "disabling_this_because_it_is_for_building_an_apk")))]
             Self::FileSystem { path: _, source: e } => e.source(),
             _ => None,
         }
@@ -270,7 +270,7 @@
     Error::Unknown { description: format!("{err}") }
 }
 
-#[cfg(all(unix, not(target_os = "ios"), not(target_os = "android")))]
+#[cfg(all(unix, not(target_os = "ios"), not(target_os = "disabling_this_because_it_is_for_building_an_apk")))]
 pub fn fs_error(path: impl Into<PathBuf>, source: std::io::Error) -> Error {
     Error::FileSystem { path: path.into(), source }
 }
@@ -393,7 +393,7 @@
     pub size: TrashItemSize,
 }
 
-#[cfg(all(not(target_os = "ios"), not(target_os = "android")))]
+#[cfg(all(not(target_os = "ios"), not(target_os = "disabling_this_because_it_is_for_building_an_apk")))]
 /// Restores all the provided [`TrashItem`] to their original location.
 ///
 /// This function consumes the provided items.
@@ -471,7 +471,7 @@
 
 #[cfg(any(
     target_os = "windows",
-    all(unix, not(target_os = "macos"), not(target_os = "ios"), not(target_os = "android"))
+    all(unix, not(target_os = "macos"), not(target_os = "ios"), not(target_os = "disabling_this_because_it_is_for_building_an_apk"))
 ))]
 pub mod os_limited {
     //! This module provides functionality which is only supported on Windows and
@@ -479,7 +479,7 @@
 
     use std::borrow::Borrow;
 
-    #[cfg(all(unix, not(target_os = "macos"), not(target_os = "ios"), not(target_os = "android")))]
+    #[cfg(all(unix, not(target_os = "macos"), not(target_os = "ios"), not(target_os = "disabling_this_because_it_is_for_building_an_apk")))]
     use std::collections::HashSet;
 
     use super::{platform, Error, TrashItem, TrashItemMetadata};
@@ -525,14 +525,14 @@
     /// # Example
     ///
     /// ```
-    /// # #[cfg(all(unix, not(target_os = "macos"), not(target_os = "ios"), not(target_os = "android")))] {
+    /// # #[cfg(all(unix, not(target_os = "macos"), not(target_os = "ios"), not(target_os = "disabling_this_because_it_is_for_building_an_apk")))] {
     /// use trash::os_limited::trash_folders;
     /// let trash_bins = trash_folders()?;
     /// println!("{trash_bins:#?}");
     /// # }
     /// # Ok::<(), trash::Error>(())
     /// ```
-    #[cfg(all(unix, not(target_os = "macos"), not(target_os = "ios"), not(target_os = "android")))]
+    #[cfg(all(unix, not(target_os = "macos"), not(target_os = "ios"), not(target_os = "disabling_this_because_it_is_for_building_an_apk")))]
     pub fn trash_folders() -> Result<HashSet<std::path::PathBuf>, Error> {
         platform::trash_folders()
     }
